test/spec/groupCtrlSpec.js

```diff
@@ -89,6 +91,51 @@ describe("Autocomplete controller", function() {
     });
   });
 
+  describe("chatChanged", function() {
+    it('if a new chat arrives, the new user name is extracted', function() {
+
+      expect(scope.response.length).to.be.eq(0);
+      expect(scope.usernames.length).to.be.eq(0);
+
+      scope.group = {}
+      scope.group.chat = [{msg: "new chat", user: "boo"}];
+      expect(scope.response.length).to.be.eq(0);
+      expect(scope.usernames.length).to.be.eq(0);
+    })
+  });
+
+  describe("addNewUser", function() {
```

Comment:
Even though it's called implicitly, `addNewUser` is still a function that can be called, so I'd have one test where you call it explicitly and write some expectations about `$scope.usernames` and `$scope.response` and one where you spy on it when calling `chatChanged` and make sure that it was called when you expect it to be called.